lib/internal/modules/esm/loader.js

```diff
@@ -230,6 +253,14 @@ class DefaultModuleLoader {
     return module.getNamespace();
   }
 
+  register(specifier, parentUrl) {
+    if (this.#delegate) {
+      return this.#delegate.register(specifier, parentUrl);
+    }
+    this.#delegate = new CustomizedModuleLoader();
+    return this.register(specifier, parentUrl);
```

Comment:
Recursion here is a bit unnecessary, isn’t it?

```suggestion
    if (!this.#delegate) {
      this.#delegate = new CustomizedModuleLoader();
    }
    return this.#delegate.register(specifier, parentUrl);
```

Reply:
Agreed!